features/page-objects/RoomSettingPage.js

```diff
@@ -13,17 +12,19 @@ class RoomSettingPage extends Page {
     await this.clickUpdateRoom();
   }
 
-  async unlock() {
-    await this.enterAccessCode()
+  async unlock(accessCode) {
+    await this.enterAccessCode(accessCode)
+
     await this.setAccessLevel('unlocked');
     await this.clickUpdateRoom();
   }
 
-  async enterAccessCode() {
+  async enterAccessCode(accessCode) {
```

Comment:
`enterAccessCode` and `setAccessCode` are named so similar it's a little confusing. `enterAccessCode` is doing more than just inputting the access code, it looks like it's also clicking submit button with `submitInput.click()`. So maybe rename `enterAccessCode` to `submitAccessCode`?

Reply:
Great catch! 

I wonder if what we _really_ want here is to decompose the responsibility for interacting with the WaitingRoom  from the RoomSettingsPage; as that I think is the root cause for the mismatched IDs that you noticed further down.

```
await waitingRoom.submitAccessCode("value");
```